index.js

```diff
@@ -0,0 +1,115 @@
+const express = require('express')
+const bodyParser = require('body-parser')
+const models = require('./models')
+
+const app = express()
+
+app.get('/movies', async (request, response) => {
+  const matchingmovies = await models.movies.findAll({
+    include: [{ model: models.directors }, { model: models.genres }]
+  })
+  return response.send(matchingmovies)
+})
+
+app.get('/movies/:identifier', async (request, response) => {
+  const matchingMovies = await models.movies.findOne({
+    where: {
+      id: request.params.identifier,
+    },
+    include: [{ model: models.directors }, { model: models.genres }]
+
+  })
+  if (matchingMovies) {
+    return response.send(matchingMovies)
+  } else {
+    return response.sendStatus(404)
+  }
+})
+
+app.get('/directors/:identifier', async (request, response) => {
+  const matchingDirector = await models.directors.findAll({
+    include: {
+      model: models.movies
+    },
+    where: {
+      id: request.params.identifier
+    }
+  })
+  if (matchingDirector) {
+    return response.send(matchingDirector)
+  } else {
+    return response.sendStatus(404)
+  }
+})
+
+app.get('/genres/:genresName', async (request, response) => {
+  const matchingGenres = await models.genres.findAll({
+    include: {
+      model: models.movies
+    },
+    where: {
+      genresName: request.params.genresName
+    }
+
+  })
+  if (matchingGenres) {
+    return response.send(matchingGenres)
+  } else {
+    return response.sendStatus(404)
+  }
+
+})
+
+app.post('/movies', bodyParser.json(), async (request, response) => {
+  const { title, directors, releaseDate, ratings, runTime, genres } = request.body
+
+  const name = directors.split(', ')
+  const genreArrays = genres.split(', ')
+
+
+  if (!title || !directors || !releaseDate || !ratings || !runTime || !genres) {
+    response.status(400).send('The following attributes are required: title, directors, releaseDate, ratings, runTime, genres')
+  } else {
+    const newMovie = await models.movies.create({ title, releaseDate, ratings, runTime })
+    response.status(201).send(newMovie)
```

Comment:
By responding here you're sending back a 201 with just the movie data to the user. This means if the other data creates fail the user will never know since they've already received a 201 response. It also means they are receiving back just the movie data and not the movie along with directors and genres.